app/services/GuidanceService.scala

```diff
@@ -20,41 +20,42 @@ import connectors.GuidanceConnector
 import javax.inject.{Inject, Singleton}
 import models.ui.Page
 import uk.gov.hmrc.http.HeaderCarrier
-
 import scala.concurrent.{ExecutionContext, Future}
+import repositories.SessionRepository
+import models.ocelot.Process
 
 @Singleton
-class GuidanceService @Inject() (connector: GuidanceConnector) {
-
-  def getStartPageUrl(processId: String)(implicit hc: HeaderCarrier, context: ExecutionContext): Future[String] = {
-    connector.getProcess(processId) map { process =>
-      PageBuilder.pages(process) match {
-        case Right(pages) =>
-          val startPage = pages.find(page => page.id == "start")
-          startPage.fold("")(page => page.url)
-
-        // TODO
-        case Left(_) => ""
-      }
+class GuidanceService @Inject() (connector: GuidanceConnector, sessionRepository: SessionRepository) {
+
+  private def startProcessView( id: String, processById: String => Future[Option[Process]])
```

Comment:
I don't know if we are only covering the happy flow here. The error handling seems to be returning a blank string. Are there plans to update this at a later date?

Reply:
Yes, maybe this should be a NOT_FOUND, will update